### PR TITLE
FIX Handle no trailing slash in other tab of archive admin

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-versioned-admin/issues/291

I've made some other changes to some [seemingly unrelated behat failures](https://github.com/silverstripe/silverstripe-versioned-admin/actions/runs/5262808189/jobs/9513413049?pr=293) that were happening only in the php 8.2 / mysql 8.0 job.  Seems like for whatever reason that "Home" page wasn't getting a page ID of 1, so I've updated the behat tests to not rely on the page ID to be 1. Normally I wouldn't do this in the same PR since it's unrelated, though even I wasn't sure if it was actually unrelated or not to the code change.

### CODE REVIEW

File: client/src/legacy/ArchiveAdmin/ArchiveAdmin.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ jQuery.entwine('ss', ($) => {
       const othersArchiveUrl = $(this).data('others-archive-url');
       if (targetClassName) {
         targetClassName = targetClassName.replace(/\\/g, '-');
-        const url = othersArchiveUrl + targetClassName;
+        const url = `${othersArchiveUrl}/${targetClassName}`;
```
Comment (ChloeSartorelli) [inline client/src/legacy/ArchiveAdmin/ArchiveAdmin.js]: This assumes `othersArchiveUrl` will now _never_ have a trailing slash.... use `joinUrlPaths()` instead per https://docs.silverstripe.org/en/5/changelogs/5.0.0/#trailing-slash

File: src/ArchiveAdmin.php [hunk 2]
```diff
@@ -282,7 +282,7 @@ public function getOtherModelSelectorField($currentModel = '')
         );
         $modelSelectField->setAttribute(
             'data-others-archive-url',
-            $this->Link('/')
+            rtrim($this->Link('/'), '/')
```
Comment (ChloeSartorelli) [inline src/ArchiveAdmin.php]: Don't do this. Respect trailing slash config.

File: tests/Behat/features/archive-other.feature [hunk 3]
```diff
@@ -0,0 +1,17 @@
+Feature: Achive admin other tab
```
Comment (ChloeSartorelli) [inline tests/Behat/features/archive-other.feature]:
```suggestion
Feature: Archive admin other tab
```
Super minor but could affect searching for "archive admin" across all files in the future so I figured I'd point it out. Won't hold back merging for it.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: client/src/legacy/ArchiveAdmin/ArchiveAdmin.js (hunk 1 above)
Comment (emteknetnz) [reply client/src/legacy/ArchiveAdmin/ArchiveAdmin.js]: Updated

File: src/ArchiveAdmin.php (hunk 2 above)
Comment (emteknetnz) [reply src/ArchiveAdmin.php]: Updated

File: tests/Behat/features/archive-other.feature (hunk 3 above)
Comment (emteknetnz) [reply tests/Behat/features/archive-other.feature]: Updated

**Review by GuySartorelli — APPROVED** [review]
Looks good and works well locally.

**Pull request merged**